lib/loginator/transaction.rb

```diff
@@ -1,21 +1,125 @@
 require 'securerandom'
 
 module Loginator
-  # Methods for generating transactional metadata.
-  module Transaction
-    def self.from_json(json_str)
-      json = MultiJson.load(json_str)
-      Loginator.const_get(json['type'].capitalize).from_hash(json)
+  # A Loginator::Transaction is meant to encapsulate a single
+  # request/response cycle.
+  class Transaction
+    class << self
+      # @param [String] json JSON body of the given transaction
+      # @return [Transaction]
+      def from_json(json)
+        obj = MultiJson.load(json, symbolize_keys: true)
+        Transaction.new(filter_hash(obj))
+      end
+
+      protected
+
+      # This is the inverse functions for the instance-level
+      # filter_hash.
+      def filter_hash(hsh)
+        [[:timestamp, ->(t) { Time.at(t) }]].map do |k, f|
+          hsh[k] = f.call(hsh[k])
+        end
+        hsh
+      end
+    end
+
+    attr_accessor :path, :request, :params, :response, :status
```

Comment:
I waffled on the attr_accessor, attr_reader, attr_writer business for a while. Right now, I'm just trying to be extremely explicit about how I think `Transactions` should be used... and attempting to convey that intent via access policies. I don't think I'm doing a very good job, but it's at least an immediately effective interface.